app/javascript/src/components/editor/CodeMirror.svelte

```diff
@@ -281,10 +285,28 @@
     })
   }
 
-  async function setScrollPosition() {
+  function setScrollPosition() {
     view.scrollDOM.scrollTo({ top: $editorScrollPositions[currentId] || 0 })
   }
 
+  function indentMultilineInserts(transaction) {
+    // Only perform this function if transaction is of an expected type performed by the user to prevent infinite loops on changes made by CodeMirror
+    if (transaction.transactions.every(tr => ["input.paste", "input.complete"].includes(tr.annotation(Transaction.userEvent)))) {
+      const [range] = transaction.changedRanges
+      const text = transaction.state.doc.toString().slice(range.fromB, range.toB)
+      const indents = getIndentForLine(view.state, range.fromB)
```

Comment:
Indentation is not all sunshine and rainbows, sadly

https://github.com/Mitcheljager/workshop.codes/assets/6181929/87e2c88f-6d4a-42b1-b462-196a1c28e0ff

I think this could benefit from reapplying indentation on each line of the pasted content?

Reply:
I made some improvements and it should be much better. It's not quite perfect but the imperfections make enough sense. VSCode has similar (but different) inconsistencies. I think this is about as close as we can get without fully parsing the text, which would be too slow for larger texts anyway.